### PR TITLE
MAINT: create utils, fix validation

### PR DESCRIPTION
**Pull request by eigenfoo**
Supersedes #77.

- [x] Create a `utils.py` with the `CHARACTER` array, model saving/loading functions, and raster function.
- [x] Generate fonts, rasterize and save them as pngs periodically during training.
- [x] Create a script to run model at inference time, generating more fonts.

Note that `fridge.py` is no more. Model saving/loading is handled by `utils.py`, along with a few more handy variables and functions.

### CODE REVIEW

File: src/infer.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    # Create random one-hot character vector
+    char_vector = torch.zeros([num_fonts, len(CHARACTERS)], device=device)
+    char_vector[range(num_fonts), range(len(CHARACTERS))] = 1
+
+    # Generate a batch of fake characters
+    fake_chars = gen(char_vector, style_vector)
+    fake_chars = torch.reshape(fake_chars, [num_fonts, -1, 2])
+    rasters = rasterize(fake_chars)
+    save_images(rasters, [resolution, resolution], path + '_raster.png')
```
Comment (eigenfoo) [inline src/infer.py]: @ccurro can you confirm that I'm using your `save_images` function properly?

Comment (eigenfoo) [reply src/infer.py]: Done.

**Comment by eigenfoo** [comment]
TODO:

- [ ] Infer with full character sets. This involves fixing matzah.
- [ ] Make sure running `infer.py` works.

File: Makefile [hunk 2]
```diff
@@ -6,6 +6,9 @@ SHELL = bash
 init:
 	find .git/hooks -type l -exec rm {} \;
 	find .githooks -type f -exec ln -sf ../../{} .git/hooks/ \;
+	mkdir output/
```
Comment (ArianaFreitag) [inline Makefile]: after we merge to master do we all need to call `make init` again?

Comment (eigenfoo) [reply Makefile]: Correct

Comment (eigenfoo) [reply Makefile]: After this merges, everyone should immediately pull. This PR changes quite a lot.

**Pull request merged**